internal/service/federatedsettingsorgconfig/resource_federated_settings_connected_org.go

```diff
@@ -149,42 +180,29 @@ func resourceMongoDBAtlasFederatedSettingsOrganizationConfigDelete(ctx context.C
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error deleting federation settings connected organization (%s): %s", federationSettingsID, err))
 	}
-
+	d.SetId("")
 	return nil
 }
 
-func resourceMongoDBAtlasFederatedSettingsOrganizationConfigImportState(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+func resourceImport(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
 	conn := meta.(*config.MongoDBClient).AtlasV2
-	federationSettingsID, orgID, err := splitFederatedSettingsOrganizationConfigImportID(d.Id())
+	federationSettingsID, orgID, err := splitImportID(d.Id())
 	if err != nil {
 		return nil, err
 	}
 
-	federatedSettingsConnectedOrganization, _, err := conn.FederatedAuthenticationApi.GetConnectedOrgConfig(context.Background(), *federationSettingsID, *orgID).Execute()
+	_, _, err = conn.FederatedAuthenticationApi.GetConnectedOrgConfig(context.Background(), *federationSettingsID, *orgID).Execute()
 	if err != nil {
 		return nil, fmt.Errorf("couldn't import Organization config (%s) in Federation settings (%s), error: %s", *orgID, *federationSettingsID, err)
 	}
 
 	if err := d.Set("federation_settings_id", *federationSettingsID); err != nil {
 		return nil, fmt.Errorf("error setting Organization config in Federation settings (%s): %s", d.Id(), err)
 	}
```

Comment:
Could we remove this code as well? Or is the read operation making use of the federation_settings_id attribute?

Reply:
The read operation only use `DecodeId()` however, we don't set `federation_settings_id` and `org_id` since they are required, therefore, we must set them here.